### PR TITLE
feat: add swipe dismiss sensitivity params for modals

### PR DESCRIPTION
**Pull request by ice-orion**
## Description

Thank u for the great lib!

The PR adds `SwipeDismissConfig` class that can be passed to `CupertinoModalSheetPage`, `CupertinoModalSheetRoute`, `ModalSheetPage` and `ModalSheetRoute` classes to configure min swipe velocity and distance for a swipe to result in a dismissal.

### CODE REVIEW

**Comment by appinteractive** [comment]
Awesome! But I think it would make more sense to add a DismissConfig (or similar) that contains the both values. At least that is how the other configs get encapsulated as far as I can tell.

File: package/lib/src/modal/swipe_dismiss_config.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      other.minFlingVelocity == minFlingVelocity &&
+      other.minDragDistance == minDragDistance;
+  }
+
+  @override
+  int get hashCode => minFlingVelocity.hashCode ^ minDragDistance.hashCode;
+
+  @override
+  String toString() => 'SwipeDismissConfig(minFlingVelocity: $minFlingVelocity, minDragDistance: $minDragDistance)';
+}
```
Comment (fujidaiti) [inline package/lib/src/modal/swipe_dismiss_config.dart]: I believe we might not need to support these methods, as they are rarely used and then just increases maintenance costs.

File: package/lib/src/modal/swipe_dismiss_config.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/// Configuration class for handling swipe-to-dismiss gestures.
+///
+/// The [SwipeDismissConfig] class provides parameters to control the behavior
+/// of swipe-to-dismiss interactions for modals. It defines the minimum velocity
+/// required for the swipe gesture ([minFlingVelocity]) and the minimum distance
+/// the user needs to drag ([minDragDistance]) for a swipe to result in a dismissal.
+class SwipeDismissConfig {
```
Comment (fujidaiti) [inline package/lib/src/modal/swipe_dismiss_config.dart]: I prefer `Sensitivity` over `Sensitivity` as I feel the latter term is a bit too abstract...

**Review by fujidaiti — COMMENTED** [review]
@ice-orion

Thank you for the contribution! It looks generally good, but there are a few points I'd like to discuss before we merge this PR.

File: package/lib/src/modal/swipe_dismiss_config.dart (hunk 2 above)
Comment (ice-orion) [reply package/lib/src/modal/swipe_dismiss_config.dart]: You probably mean `Sensitivity` over `Swipe`? Do you suggest to rename the class to something like `DismissSensitivityConfig`? If not, could you please elaborate your point

Comment (fujidaiti) [reply package/lib/src/modal/swipe_dismiss_config.dart]: Sorry, that was a typo. I meant that a name like `SwipeDismissSensitivity` might be preferable to `SwipeDismissConfig`, as the word `Config` is a rather broad term.

File: package/lib/src/modal/swipe_dismiss_config.dart (hunk 1 above)
Comment (ice-orion) [reply package/lib/src/modal/swipe_dismiss_config.dart]: yea, removed

File: package/lib/src/modal/swipe_dismiss_config.dart (hunk 2 above)
Comment (ice-orion) [reply package/lib/src/modal/swipe_dismiss_config.dart]: Got it! Renamed the class and the corresponding file.

**Comment by fujidaiti** [comment]
Can you also make the following changes?

- Bump the version to `0.10.0` in pubspec.yaml and update CHANGELOG file
- Add an example of `SwipeDismissSensitivity` to `tutorial/imperative_modal_route.dart` or somewhere
- Run `dart format .` to ensure the CI passes

**Comment by fujidaiti** [comment]
@ice-orion

I renamed `minFlingVelocity` to `minFlingVelocityRatio` to reflect its actual meaning and changed the default value from 1.0 to 2.0. Could you test the new default configuration to see if it's okay?

**Comment by fujidaiti** [comment]
@ice-orion Thank you! I'll resolve the conflicts and add some tests this weekend.

**Comment by fujidaiti** [comment]
@ice-orion I changed the default minDragDistance to 200, as 300 is a bit too large, especially for small sheets.

**Pull request merged**